### PR TITLE
build: support local compilation with cl_intel.h under libxcam

### PR DESCRIPTION
**Pull request by liuyinhangx**
* if autoconfig can not find cl_intel.h in system directory,
   then download cl_intel.h to ext directory

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -126,6 +126,17 @@ if test "$HAVE_LIBCL" -eq 1; then
     if test "x$GAWK" = "xno"; then
         AC_MSG_ERROR([gawk not found])
     fi
+
+    HAVE_CL_INTEL_H=1
+    AC_CHECK_HEADERS([CL/cl_intel.h], [HAVE_CL_INTEL_H=1], [HAVE_CL_INTEL_H=0])
+    if test "$HAVE_CL_INTEL_H" -eq 0; then
+        XCAM_WGET(
+            [https://raw.githubusercontent.com/intel/beignet/Release_v1.3/include/CL/cl_intel.h],
```
Comment (windyuan) [inline configure.ac]: why not https://cgit.freedesktop.org/beignet/tree/include/CL/cl_intel.h?h=Release_v1.3

File: .gitignore [hunk 2]
```diff
@@ -15,7 +15,8 @@ config.*
 /missing
 /stamp-h1
 /pkgconfig/*.pc
-/m4
+/m4/*
+!/m4/xcam-utils.m4
```
Comment (windyuan) [inline .gitignore]: maybe you can add another file m4/.gitignore with 
libtool.m4
lt~obsolete.m4
ltoptions.m4
ltsugar.m4
ltversion.m4

File: configure.ac (hunk 1 above)
Comment (liuyinhangx) [inline configure.ac]: The correct url is: https://cgit.freedesktop.org/beignet/plain/include/CL/cl_intel.h?h=Release_v1.3,
but the cl_intel.h file from this url has timestamp issue, which will cause repetitive compilation.

Comment (windyuan) [inline configure.ac]: then need check if file not exit or md5sum failed. then download the file.
and also need check wget return values. 0 => success, else report a error msg.

Comment (liuyinhangx) [inline configure.ac]: these functionality are implemented in macro XCAM_WGET.

**Comment by liuyinhangx** [comment]
update patch, please help to review, thanks.

**Comment by windyuan** [comment]
merged with changes. thanks.

commit 789a8e8186e789106b76e9cb3be664895860e9a2
commit 4bd7e36f95da9ea5563254107773824175426762

		diff --git a/m4/xcam-utils.m4 b/m4/xcam-utils.m4
		index 5a2fa5c..2de7deb 100644
		--- a/m4/xcam-utils.m4
		+++ b/m4/xcam-utils.m4
		@@ -15,16 +15,16 @@ AC_DEFUN([XCAM_WGET],
			 MD5_CORRECT=yes
			 XCAM_MD5SUM([$2], [$3], [MD5_CORRECT=yes], [MD5_CORRECT=no])
			 AS_IF([test "x$MD5_CORRECT" = "xyes"],
		-        [AC_MSG_NOTICE([checking $2... yes])],
		+        [AC_MSG_NOTICE([checking $2 md5sum ... ok])],
				 [
		-            AC_MSG_NOTICE([checking $2... no])
		+            AC_MSG_NOTICE([downloading $2...])
					 dir=`dirname $2`
					 AS_IF([test ! -d $dir], [mkdir -p $dir])

		-            wget --tries=2 --timeout=5 --no-use-server-timestamps $1 -O $2
		-            AS_IF([test "$?" != 0], [AC_MSG_ERROR([wget: get $2 failed])])
		+            wget --tries=2 --timeout=5 -q --no-use-server-timestamps $1 -O $2
		+            AS_IF([test "$?" != 0], [AC_MSG_ERROR([download/wget $2 failed])])

		-            XCAM_MD5SUM([$2], [$3], , [AC_MSG_ERROR([$2: check md5sum failed])])
		+            XCAM_MD5SUM([$2], [$3], [AC_MSG_NOTICE([checking $2 md5sum ... ok])], [AC_MSG_ERROR([checking $2 md5sum ... failed])])
				 ])
		 ])

**Pull request closed without merging**